### ISSUE TITLE
Cue points for track 1 index 02+ are based on index 01

### ISSUE THREAD

**Issue by hcs64** [saramibreak/DiscImageCreator#172]
**Version**
20221217

**Describe the bug**
On Namco Video Game Graffiti Vol. 2, the split track cuesheet has track 1 indexes 02 and above at different time than in the whole disc image. They're off by 37 frames, which is index 01, so seems that the indexes are relative to index 01 instead of the beginning of the track. This seems to only be an issue on track 1 when index 01 is past 00:00:00.

This seems to be rare, I've not found another example of a disc like this.

**Screenshots**

tracks cue:
```
FILE "VDR-5222 (Track 1).bin" BINARY
  TRACK 01 AUDIO
    INDEX 00 00:00:00
    INDEX 01 00:00:37
    INDEX 02 00:11:40
    INDEX 03 00:37:33
    INDEX 04 03:11:33
    INDEX 05 03:40:68
    INDEX 06 06:07:28
    INDEX 07 06:35:50
    INDEX 08 07:52:45
    INDEX 09 08:33:33
    INDEX 10 10:11:38
    INDEX 11 10:54:55
    INDEX 12 12:55:43
    INDEX 13 13:49:40
    INDEX 14 15:17:53
    INDEX 15 19:33:48
    INDEX 16 24:40:38
    INDEX 17 25:34:50
```

img.cue:
```
FILE "VDR-5222.img" BINARY
  TRACK 01 AUDIO
    INDEX 00 00:00:00
    INDEX 01 00:00:37
    INDEX 02 00:12:02
    INDEX 03 00:37:70
    INDEX 04 03:11:70
    INDEX 05 03:41:30
    INDEX 06 06:07:65
    INDEX 07 06:36:12
    INDEX 08 07:53:07
    INDEX 09 08:33:70
    INDEX 10 10:12:00
    INDEX 11 10:55:17
    INDEX 12 12:56:05
    INDEX 13 13:50:02
    INDEX 14 15:18:15
    INDEX 15 19:34:10
    INDEX 16 24:41:00
    INDEX 17 25:35:12
```

**Disc title**
Namco Video Game Graffiti Vol. 2
Cat. no. VDR-5222

**Disc ringcode**
`VDR-5222-1-A1E`

**URL**
[Amazon](https://www.amazon.com/): https://a.co/d/beEJ4qY

**Log file**
[VDR-5222_dic_test_20221217.zip](https://github.com/saramibreak/DiscImageCreator/files/10326433/VDR-5222_dic_test_20221217.zip)

**Comment by saramibreak** [comment]
It's correct. MSF of _tracks cue_ uses relative time, while MSF of _img.cue_ uses absolute time.

**Comment by superg** [comment]
sarami, you're correct with this statement.
But the problem here is that this disc TOC track 1 starts from lba 37 (index 1), but when you do a track split, you start with lba 0.
Due to 37 index 0 sectors prepended to Track 1, your index timepoints are all shifted 37 sectors in relation to track 1.
In the past, I had a discussion with other mods, where should track 1 split start if it's not LBA 0 in toc, and Fireball responded that we should always start from lba 0, which DIC does correctly, just the index entries have to be corrected.

**Comment by hcs64** [comment]
Thanks sarami, let me try to explain better. I think the img .cue is correct to say index 1 goes from 00:00:37 to 00:12:02. But this is inconsistent with the track .cue, where index 1 goes from 00:00:37 to 00:11:40, unless some frames are missing from index 1 in the track .bin (I don't think that is the case).

For track 1, `nLBAofFirstIdx` [is set](https://github.com/saramibreak/DiscImageCreator/blob/f1b1ece01445e2dbc026ba49c8dd4f75e3eaf764/DiscImageCreator/output.cpp#L2131-L2138) to the start of index 1, but there's some index 0 sectors in track 1 on this disc. You have [a special case](https://github.com/saramibreak/DiscImageCreator/blob/f1b1ece01445e2dbc026ba49c8dd4f75e3eaf764/DiscImageCreator/output.cpp#L2169) to output indexes 0 and 1, but [later indexes](https://github.com/saramibreak/DiscImageCreator/blob/f1b1ece01445e2dbc026ba49c8dd4f75e3eaf764/DiscImageCreator/output.cpp#L2231) are relative to `nLBAofFirstIdx`, which is the start of index 1, past the start of the track.

So this only occurs when track 1 contains both index 0 and 2.

**Comment by saramibreak** [comment]
> MSF of tracks cue uses relative time

Ok, correctly, MSF of _tracks cue_ uses absolute time from **the each track**, not relative time.

**Comment by saramibreak** [comment]
test version 
[DiscImageCreator_test.zip](https://github.com/saramibreak/DiscImageCreator/files/10328204/DiscImageCreator_test.zip)

**Comment by hcs64** [comment]
That looks good for track 1!

But, now track 3 looks like this in the tracks .cue:

```
FILE "VDR-5222 (Track 3).bin" BINARY
  TRACK 03 AUDIO
    INDEX 01 00:00:00
    INDEX 02 35:11:47
    INDEX 03 37:21:47
    INDEX 04 40:37:65
    INDEX 05 41:14:55
    INDEX 06 41:47:50
```

It looks like index 2+ is absolute. I think these were correct before in 20221217:

```
FILE "VDR-5222 (Track 3).bin" BINARY
  TRACK 03 AUDIO
    INDEX 01 00:00:00
    INDEX 02 04:44:27
    INDEX 03 06:54:27
    INDEX 04 10:10:45
    INDEX 05 10:47:35
    INDEX 06 11:20:30
```

Thanks again, happy new year!

**Comment by saramibreak** [comment]
fixed
[DiscImageCreator_test.zip](https://github.com/saramibreak/DiscImageCreator/files/10328378/DiscImageCreator_test.zip)

**Comment by saramibreak** [comment]
fixed
[DiscImageCreator_test.zip](https://github.com/saramibreak/DiscImageCreator/files/10328385/DiscImageCreator_test.zip)

**Comment by hcs64** [comment]
This looks good for VDR-5222, thanks!

But the problem has moved to other tracks with index 0 and 2:

Battle Gear: The Discovery of Roads (ZTTL-0054, [redump](http://redump.org/disc/94322/), [Amazon](https://a.co/d/5AWA1Yq)) has index 0-3 on track 10. With the test above (20230101T151119, [logs](https://github.com/saramibreak/DiscImageCreator/files/10328475/ZTTL-0054_dic_issue172-2_logs.zip)) this gives:

```
FILE "ZTTL-0054 (Track 10).bin" BINARY
  TRACK 10 AUDIO
    INDEX 00 00:00:00
    INDEX 01 00:03:22
    INDEX 02 00:12:39
    INDEX 03 02:43:12
```

I think it was correct before as:

```
FILE "ZTTL-0054 (Track 10).bin" BINARY
  TRACK 10 AUDIO
    INDEX 00 00:00:00
    INDEX 01 00:03:22
    INDEX 02 00:09:17
    INDEX 03 02:39:65
```

That was consistent with the img.cue:

```
  TRACK 10 AUDIO
    INDEX 00 17:49:05
    INDEX 01 17:52:27
    INDEX 02 17:58:22
    INDEX 03 20:28:70
```

Index 2 is 00:09:17 (692 frames) past index 0.

**Comment by saramibreak** [comment]
fixed 
[DiscImageCreator_test.zip](https://github.com/saramibreak/DiscImageCreator/files/10328967/DiscImageCreator_test.zip)

**Comment by hcs64** [comment]
Looks good for everything I tested! VDR-5222 now seems correct, and everything else correctly matches the older versions.

Beyond VDR-5222 and ZTTL-0054 mentioned above, I tested [INTIR-013](http://redump.org/disc/94760/) (Rockman 9 OST, many indexes including 0 on tracks 34 and 35) and [PSCX-1005](http://redump.org/disc/94645/) (Final Fantasy III Legend of the Eternal Wind, index 1-2 on track 1). If you want the subchannel logs for any of these let me know.
